Saplings-Projects/1M_sub
# Issue-46: Map Display

# Description

Button for Map Icon on main screen and display of map and rooms that has been generated in #30 

## Related issue(s)

Related to Issue #46 

## List of changes

A more detailed list of the changes.
- Creation of MapUI Scene. This will be used for displaying the Map. Has a return button to go back to the main screen
    - Added logic to generate the rooms dynamically and scroll the map if need be. 
- Addition of Map Icon to TestingScene to open up the Map UI
- Update Map Manager to hold the array of floors, and to create the 2D map when Map Manager is instantiated
    - Add Check if map has been instantiated
- Update RoomBase to return the abbreviation of the type of room it is (used just for testing)
- Added MapManager to autoloader in order to access the map_array within MapUI scene

## Tests

Kind of stumped as to what to test here, some recommendations are welcome!

## Additional notes

Vids of the map in different sizes:

https://github.com/Saplings-Projects/1M_sub/assets/5923102/1e1ee418-9107-4fea-991e-12a8e79d2efd

https://github.com/Saplings-Projects/1M_sub/assets/5923102/a29c321f-8cde-41a8-9241-61f7b0521d92
